src/main/java/org/opensearch/knn/index/codec/transfer/VectorTransferByteToFloat.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.transfer;
+
+import org.opensearch.knn.index.codec.util.SerializationMode;
+import org.opensearch.knn.jni.JNICommons;
+
+import java.io.ByteArrayInputStream;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Vector transfer for byte vectors by casting them as float vectors
+ */
+public class VectorTransferByteToFloat extends VectorTransfer {
```

Comment:
This class is unnecessary. Already have classes to transfer bytes and floats to JNI layer. Let reuse those classes to do so. 

If we have to do this, it means the interface of VectorTransfer is not solid enough and needs more thought process.